### ISSUE TITLE
Assignment of a reference variable should be consider a use of that variable

### ISSUE THREAD

**Issue by alexpott** [sirbrillig/phpcs-variable-analysis#126]
Consider the following PHP code:
```php
<?php

/**
 * @file
 */

class A {
  protected $prop = [];

  public function __construct() {
    $this->prop[] = 'foo';
  }

  public function &getProp() {
    return $this->prop;
  }

}

function doit() {
  $a = new A();

  $var = &$a->getProp();
  $var = ['bar'];
  return $a;
}

```

Produces a warning:
```
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 23 | WARNING | Unused variable $var.
----------------------------------------------------------------------
```

But the variable is used to reset the A->prop is being changed and the class is then returned. I think in the case of references we need to consider assignment as the variable being used.

**Comment by sirbrillig** [comment]
🤔 I don't think I've ever seen [return references](https://www.php.net/references.return) used before! Thanks for the example case. I think that my understanding of assignment on reference variables was lacking; I've just read up on the docs and learned a few things. This is definitely a false positive.

**Comment by sirbrillig** [comment]
Fixed by #128 (I think! If I'm wrong, please re-open.)
